### PR TITLE
feat(atenlib): topk; bool argument type annotation; node of multiple outputs

### LINKED ISSUE #283: Support bool as an attribute type
#281 

```python
@script()
def aten_logsumexp(self: TReal, dim: INT64["M"], keepdim: bool = False) -> TReal:
    # logsumexp(Tensor self, int[1] dim, bool keepdim=False) -> Tensor

    return op.ReduceLogSumExp(self, dim, keepdims=keepdim)
```

produces

ERROR onnxscript/test/function_libs/torch_aten/ops_correctness_test.py - TypeError: value "<onnxscript.values.Dynamic object at 0x7f2611f68640>" is not valid attribute data type.

Even though `dim` should be an input?

cc @gramalingam @fatcat-z @xiaowuhu

### PR DESCRIPTION
**Pull request by BowenBao**
~~Concern on the annotation of `k` as `int`. I assume this would imply static value in onnx/onnxscript? Since scalars could be non-static in torch, would we want a scalar type wrapper too?~~ Answer is use `INT64`.

* Add `aten_topk`.
* Extend onnx script type annotation for bool argument for attributes.
* Extend opinfo test for multiple outputs.

Fixes #283

### CODE REVIEW

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 1]
```diff
@@ -4741,7 +4741,16 @@ def aten_topk(
 ) -> tuple[TensorType, TensorType]:
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/core.py:4741]: need: 

```
@torch_op("aten::topk")
```

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 2]
```diff
@@ -4741,7 +4741,16 @@ def aten_topk(
...
     # topk(Tensor self, int k, int dim=-1, bool largest=True, bool sorted=True) -> (Tensor values, Tensor indices)
 
-    raise NotImplementedError()
+    # TODO: Does this support dynamic k value?
+    is_scalar_input = op.Size(op.Shape(self)) == 0
+    if is_scalar_input:
+        self = op.Unsqueeze(self, op.Constant(value_ints=[0]))
+    k_value = op.Constant(value_ints=[k])
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/core.py:4748]:
```suggestion
```

**Comment by justinchuby** [comment]
Errors related to #298 are fine to ignore

File: onnxscript/function_libs/torch_aten/ops/core.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py:4741]: self: TReal, … -> tuple[TReal, INT64]

Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py:4741]: It does both compilation and registration. however this registry is only for code gen and not public facing.

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 """
 from __future__ import annotations
 
-from typing import Any, Optional, Sequence, Union
+from typing import Any, Optional, Sequence, Union, Tuple
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/core.py:14]:
```suggestion
from typing import Any, Optional, Sequence, Union
```

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 4]
```diff
@@ -4736,18 +4736,18 @@ def aten_to_sparse_csr(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
+@torch_op("aten::topk", trace_only=True)
 def aten_topk(
-    self: TensorType, k: int, dim: int = -1, largest: bool = True, sorted: bool = True
-) -> tuple[TensorType, TensorType]:
+    self: TReal, k: INT64, dim: int = -1, largest: bool = True, sorted: bool = True
+) -> Tuple[TReal, INT64]:
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/core.py:4742]:
```suggestion
) -> tuple[TReal, INT64]:
```

nit: tuple is good since we imported the future

**Review by justinchuby — APPROVED** [review]

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 5]
```diff
@@ -4736,12 +4736,21 @@ def aten_to_sparse_csr(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
+@torch_op("aten::topk", trace_only=True)
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/core.py:4739]: Can this be compiled?

**Review by justinchuby — CHANGES_REQUESTED** [review]
wait: compilation

File: onnxscript/test/function_libs/torch_aten/ops_correctness_test.py [hunk 6]
```diff
@@ -508,26 +510,35 @@ def test_output_match(self, device: str, dtype: torch.dtype, op):
...
-                    torch.tensor(function_output),
-                    torch.tensor(torch_output),
-                    rtol=rtol,
-                    atol=atol,
-                )
+                # TODO: add pytree structure comparison.
+                flattened_torch_outputs, _ = pytree.tree_flatten(torch_output)
+                flattened_function_outputs, _ = pytree.tree_flatten(function_output)
+
+                assert len(flattened_torch_outputs) == len(flattened_function_outputs)
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply onnxscript/test/function_libs/torch_aten/ops_correctness_test.py:517]: aka. assert len() > 0

File: onnxscript/function_libs/torch_aten/ops/core.py (hunk 4 above)
Comment (BowenBao) [reply onnxscript/function_libs/torch_aten/ops/core.py:4760]: got this error when it was lower case tuple
```
______________________________________________________________________________________________________ ERROR collecting onnxscript/test/function_libs/torch_aten/ops_correctness_test.py ______________________________________________________________________________________________________
onnxscript/test/function_libs/torch_aten/ops_correctness_test.py:18: in <module>
    from onnxscript.function_libs.torch_aten.ops import core as core_ops
onnxscript/function_libs/torch_aten/ops/core.py:4758: in <module>
    def aten_topk(
onnxscript/function_libs/torch_aten/registration.py:74: in wrapper
    processed_func = onnxscript.script()(func)
onnxscript/main.py:107: in transform
    result = script_check(ast, opset, env, src, default_opset=default_opset)
onnxscript/main.py:53: in script_check
    return convert.top_level_stmt(f)
onnxscript/converter.py:1357: in top_level_stmt
    fn_ir = self.translate_function_def(stmt)
onnxscript/converter.py:1331: in translate_function_def
    type_annotation = self.eval_constant_expr(fn.returns)
onnxscript/converter.py:412: in eval_constant_expr
    return eval(cpl, self.globals, locals)  # noqa: DUO104
<ast>:4: in <module>
    ???
E   TypeError: 'type' object is not subscriptable
=================================================================================================================================== short test summary info ===================================================================================================================================
ERROR onnxscript/test/function_libs/torch_aten/ops_correctness_test.py - TypeError: 'type' object is not subscriptable
```

File: onnxscript/function_libs/torch_aten/ops/core.py (hunk 5 above)
Comment (BowenBao) [reply onnxscript/function_libs/torch_aten/ops/core.py]: Done. Thx for tip on `input_wrangler`. Do we have an issue tracking the attr-kwargs-onnxscript limitation?

**Comment by BowenBao** [comment]
CI fails on `AttributeError: module 'onnx.parser' has no attribute 'parse_function'`. Guess I'm using a newer ONNX version locally :(

File: onnxscript/function_libs/torch_aten/ops/core.py (hunk 4 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py:4760]: Ahh ok I see what's going on. Tuple is good then

File: onnxscript/function_libs/torch_aten/ops/core.py (hunk 5 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py]: Just added https://github.com/microsoft/onnx-script/issues/305

**Comment by justinchuby** [comment]
Could you add a skip on onnx version for now?

**Comment by justinchuby** [comment]
Yes - the nightly build passes but the 1.13 release fails

**Review by justinchuby — APPROVED** [review]
lgtm!

**Comment by justinchuby** [comment]
lintrunner -a is needed too

File: onnxscript/test/type_annotation_test.py [hunk 7]
```diff
@@ -66,6 +69,10 @@ def unknown_rank(A: FLOAT[...], B: FLOAT[...]) -> FLOAT[...]:
         with self.assertRaises(ValueError):
             FLOAT[10][20]  # Invalid usage. pylint: disable=pointless-statement
 
+    @unittest.skipIf(
+        Version(onnx.__version__) < Version("1.13"),
```
Comment (justinchuby) [inline onnxscript/test/type_annotation_test.py:73]: <=

Comment (BowenBao) [reply onnxscript/test/type_annotation_test.py:73]: 1.13 works for this one

Comment (justinchuby) [reply onnxscript/test/type_annotation_test.py:73]: ahh I see. Great thanks

**Pull request merged**